src/Engine/Spells/Spells.cpp

```diff
@@ -33,253 +33,261 @@ std::array<TownPortalData, 6> TownPortalList =  // 4ECBB8
-      {SPRITE_SPELL_EARTH_TELEKINESIS, 0},
-      {SPRITE_SPELL_EARTH_DEATH_BLOSSOM, 0},
-      {SPRITE_SPELL_EARTH_MASS_DISTORTION, 0},
-
-      {SPRITE_SPELL_SPIRIT_DETECT_LIFE, 0},
-      {SPRITE_SPELL_SPIRIT_BLESS, 0},
-      {SPRITE_SPELL_SPIRIT_FATE, 0},
-      {SPRITE_SPELL_SPIRIT_TURN_UNDEAD, 0},
-      {SPRITE_SPELL_SPIRIT_REMOVE_CURSE, 0},
-      {SPRITE_SPELL_SPIRIT_PRESERVATION, 0},
-      {SPRITE_SPELL_SPIRIT_HEROISM, 0},
-      {SPRITE_SPELL_SPIRIT_SPIRIT_LASH, 0},
-      {SPRITE_SPELL_SPIRIT_RAISE_DEAD, 0},
-      {SPRITE_SPELL_SPIRIT_SHARED_LIFE, 0},
-      {SPRITE_SPELL_SPIRIT_RESSURECTION, 0},
-
-      {SPRITE_SPELL_MIND_REMOVE_FEAR, 0},
-      {SPRITE_SPELL_MIND_MIND_BLAST, 0},
-      {SPRITE_SPELL_MIND_PROTECTION_FROM_MIND, 0},
-      {SPRITE_SPELL_MIND_TELEPATHY, 0},
-      {SPRITE_SPELL_MIND_CHARM, 0},
-      {SPRITE_SPELL_MIND_CURE_PARALYSIS, 0},
-      {SPRITE_SPELL_MIND_BERSERK, 0},
-      {SPRITE_SPELL_MIND_MASS_FEAR, 0},
-      {SPRITE_SPELL_MIND_CURE_INSANITY, 0},
-      {SPRITE_SPELL_MIND_PSYCHIC_SHOCK, 0},
-      {SPRITE_SPELL_MIND_ENSLAVE, 0},
-
-      {SPRITE_SPELL_BODY_CURE_WEAKNESS, 0},
-      {SPRITE_SPELL_BODY_FIRST_AID, 0},
-      {SPRITE_SPELL_BODY_PROTECTION_FROM_BODY, 0},
-      {SPRITE_SPELL_BODY_HARM, 0},
-      {SPRITE_SPELL_BODY_REGENERATION, 0},
-      {SPRITE_SPELL_BODY_CURE_POISON, 0},
-      {SPRITE_SPELL_BODY_HAMMERHANDS, 0},
-      {SPRITE_SPELL_BODY_CURE_DISEASE, 0},
-      {SPRITE_SPELL_BODY_PROTECTION_FROM_MAGIC, 0},
-      {SPRITE_SPELL_BODY_FLYING_FIST, 0},
-      {SPRITE_SPELL_BODY_POWER_CURE, 0},
-
-      {SPRITE_SPELL_LIGHT_LIGHT_BOLT, 0},
-      {SPRITE_SPELL_LIGHT_DESTROY_UNDEAD, 0},
-      {SPRITE_SPELL_LIGHT_DISPEL_MAGIC, 0},
-      {SPRITE_SPELL_LIGHT_PARALYZE, 0},
-      {SPRITE_SPELL_LIGHT_SUMMON_ELEMENTAL, 0},
-      {SPRITE_SPELL_LIGHT_DAY_OF_THE_GODS, 0},
-      {SPRITE_SPELL_LIGHT_PRISMATIC_LIGHT, 0},
-      {SPRITE_SPELL_LIGHT_DAY_OF_PROTECTION, 0},
-      {SPRITE_SPELL_LIGHT_HOUR_OF_POWER, 0},
-      {SPRITE_SPELL_LIGHT_SUNRAY, 0},
-      {SPRITE_SPELL_LIGHT_DIVINE_INTERVENTION, 0},
-
-      {SPRITE_SPELL_DARK_REANIMATE, 0},
-      {SPRITE_SPELL_DARK_TOXIC_CLOUD, 0},
-      {SPRITE_SPELL_DARK_VAMPIRIC_WEAPON, 0},
-      {SPRITE_SPELL_DARK_SHRINKING_RAY, 0},
-      {SPRITE_SPELL_DARK_SHARPMETAL, 0},
-      {SPRITE_SPELL_DARK_CONTROL_UNDEAD, 0},
-      {SPRITE_SPELL_DARK_PAIN_REFLECTION, 0},
-      {SPRITE_SPELL_DARK_SACRIFICE, 0},
-      {SPRITE_SPELL_DARK_DRAGON_BREATH, 0},
-      {SPRITE_SPELL_DARK_ARMAGEDDON, 0},
-      {SPRITE_SPELL_DARK_SOULDRINKER, 0},
-
-      { SPRITE_ARROW_PROJECTILE, 0},
-      { SPRITE_ARROW_PROJECTILE, 0},
-      { SPRITE_BLASTER_PROJECTILE, 0}}};
-
-SpellData::SpellData(int16_t innormalMana, int16_t inExpertLevelMana,
-                     int16_t inMasterLevelMana, int16_t inMagisterLevelMana,
+/**
+ * @offset 0x4E3ACC
+ */
+std::array<SPRITE_OBJECT_TYPE, SPELL_ANY_WITH_SPRITE_COUNT> SpellSpriteMapping = {{
```

Comment:
Since we're touching this, then let's convert it into an `IndexedArray`.

Reply:
I changed but to index it need SPELL_TYPE now.
Wanted to change uSpellID type to SPELL_TYPE in CastSpellInfo but then checker for sizeof started to fail.
I left a TODO comment for now.

Reply:
OK!